src/com/lundekhan/stokk/StokkHelper.kt

```diff
@@ -0,0 +1,24 @@
+package com.lundekhan.stokk
+
+import kotlin.math.pow
+
+object StokkHelper {
+    fun compoundInterest(years: Int, startAmount: Double, interest: Double, monthlyDeposit: Int): Double =
+        (0 until years*12).fold(startAmount) { acc, _ -> (acc + monthlyDeposit) * interest.pow(1.0/12) }
+
+    fun compoundInterestMonthByMonth(years: Int, startAmount: Double, interest: Double, monthlyDeposit: Int): List<Double> =
+        (0 until years*12).fold(listOf(startAmount)) { acc, index -> acc + (acc[index] + monthlyDeposit) * interest.pow(1.0/12) }
```

Comment:
There's a new function in town called `scan` which solves this for you, amazing right? :D

Reply:
same with 1/12

Reply:
Should we use experimental stuff? And I am not sure how it helps me. Seems to be very related to fold. Just that i do not have to add to an list?

Reply:
Not sure, test it out just to learn it :) Exists in other languages too.

Reply:
`scan` should be used <3

Reply:
Example
```kotlin
@ExperimentalStdlibApi
object StokkHelper {
    const val month: Double = 1.0 / 12
    fun compoundInterest(years: Int, startAmount: Double, interest: Double, monthlyDeposit: Int): Double =
        (0 until years*12).fold(startAmount) { acc, _ -> (acc + monthlyDeposit) * interest.pow(month) }

    fun compoundInterestMonthByMonth(years: Int, startAmount: Double, interest: Double, monthlyDeposit: Int): List<Double> =
        (0 until years*12).scan(startAmount) { acc, _ -> (acc + monthlyDeposit) * interest.pow(month) }

    // Return amount of months you can withdraw X amount of money each month
    fun amountOfMonthsWithdrawal(startAmount: Double = 0.0, interest: Double = 1.0, monthlyWithdrawal: Int = 0): Int {
        var month = 0
        var amount = startAmount
        while(amount > 0){
            amount -= monthlyWithdrawal.toDouble()
            amount *= interest.pow(month)
            month += 1
        }
        return month
    }
```